app/Http/Controllers/AjaxController.php

```diff
@@ -46,15 +49,64 @@ public function processFriendRequest(Request $req) {
                 if ($req->accepted == 'accepted') {
                     $friendship->status_code = 'accepted';
                     $friendship->save();
+                    /*friend request now says has accepted your friend request*/
+                    NotificationManager::deleteFriendRequestNotification($friendship);
+                    NotificationManager::makeFriendAcceptNotification($friendship);
                     return response()->json(array("message" => "friend request accepted!"), 200);
                 } else if ($req->accepted == 'declined') {
                     $friendship->status_code = 'declined';
                     $friendship->save();
+                    /*set previous friend req notification to denied*/
+                    NotificationManager::deleteFriendRequestNotification($friendship);
                     return response()->json(array("message" => "friend request denied!"), 200);
                 }
             }
             // accepted is a temp name; may change
         }
         return response()->json(array("message" => "generic error message"), 500);
     }
+
+    /**
+     * Get notifications where user is the recipient of the request.
+     * TODO currently only FRIEND notifications: add expense notifications too
+     * TODO set a cap on how many notifications to pull.
+     * @return an array of arrays. Each sub array contains 1. Notification object, 2. Notification message (string)
+     */
+    public function getNotifications(Request $req)
+    {
+        if($req->ajax()) {
+            $requests = DB::table('notifications')
+                ->where('recipient_username', Auth::user()->username)
+                ->where('category', 2) //category 2 is for just Friend notifications
+                //->where('type', 1) //Just request notifications
+                ->orderBy('is_read', 'ASC')
```

Comment:
actually this makes sense too. like we want the notifications that are haven't read yet to be on top. just leave it like this for now and then let's judge after